packages/snaps-utils/src/manifest/validator-types.ts

```diff
@@ -0,0 +1,44 @@
+import type { SnapFiles, UnvalidatedSnapFiles } from '../types';
+import type { SnapManifest } from './validation';
+
+// TODO(ritave): Research using patch based fixing similar to eslint
+//               https://eslint.org/docs/latest/extend/custom-rules#applying-fixes
+export type ValidatorFix = (files: {
+  manifest: SnapManifest;
+}) => { manifest: SnapManifest } | Promise<{ manifest: SnapManifest }>;
```

Comment:
You could define a helper type for this to reduce duplication.

```ts
type Promisable <Type> = Type | Promise<Type>;
type ValidatorFix = (/* ... */) => Promisable<{ manifest: SnapManifest }>
```

Reply:
Potentially
```ts
type ValidatorFix = (/* ... */) => Promisable<SnapManifest>
```

Reply:
I'd prefer to leave it as is, seems easier to read than hiding it inside a layer of abstraction

Reply:
I don't think it makes it that less readable? `Promisable` seems like a pretty recognisable term.